### PR TITLE
LPS-61795 Local staging permissions

### PR DESCRIPTION
**Pull request by topolik**
Hi @brianchandotcom,

This is a second part of local staging permissions fixes you reviewed a couple months ago (https://github.com/topolik/liferay-portal/pull/228), you asked to move it into a module. You merged the first part here: https://github.com/brianchandotcom/liferay-portal/pull/34136

In local staging we use always live groupId for permission checking.
=> Based on a discussion with Ray, any Group related roles and site/organization/UserGroup assignments must be done using live groupId, otherwise permission framework won't find them.

(1) The core idea is to make sure API consumers use live groupId for roles assignments and site/organization/UserGroup/Role memberships. 
(2) Next idea is to harden portal framework instead of fixing portlets and API consumers.

A commit by commit explanation:

**LPS-61795 Move existing user/organization/user-group assignments (9670498)**

A VerifyProces that rewrites existing staging groupIds into live groupIds in Organization,Site,UserGroup and Role.

**LPS-61795 Fix ULSI to allow advices to intercept the calls (359e786)**

Make sure the call goes through advices stack

**LPS-61795 Create Spring advices around portal API (98291a0)**

Advices to replace groupId in portal API calls.

`LiveGroupStagingAdvice` is a parent for all advices which:
1, when invoked, iterates over registered *LocalService methods to replace the groupId, if needed
2, registers *LocalService methods that were generated by ServiceBuilder and are related to Group entity
3, supports registering of custom methods that developers introduced

The other `*StagingAdvices` implementations provide details about respective LocalService methods to register.

If a method doesn't exist an exception is thrown in runtime during advices initialization.

**LPS-61795 Poor man coverage check to warn developers (b82d761)**

Processes the related `LocalService`s and checks if there is a method we forget to cover or is a new one.

Also throws a runtime exception during advices initialization.

**LPS-61795 (1) First remove roles relations and after then the group … (a29c1cf)**

Bug fix - avoid NPE caused by incorrect order of entities removal.

**LPS-61795 Register Spring Advices onto corresponding Spring services (5113c01)**

This was a hard one because we don't have any support for portal advices in modules.

Basically hooks into each `LocalService` proxy to add the new advice.

**SF by Mate**

Made my code almost perfect :)

HTH. Thanks!

### CODE REVIEW

**Comment by juangon** [comment]
Don't worry about that failing test, fix is already in master: https://github.com/liferay/liferay-portal/commit/e23acc2fd698f75fd14caa1f9c3d6bbb15934160

**Comment by brianchandotcom** [comment]
@topolik any way to break this up into smaller pulls? Still 1k lines ;)

**Comment by topolik** [comment]
I try to.

**Comment by topolik** [comment]
First part: https://github.com/brianchandotcom/liferay-portal/pull/35858

**Comment by topolik** [comment]
Second part: https://github.com/brianchandotcom/liferay-portal/pull/35954

**Comment by topolik** [comment]
Third part: https://github.com/brianchandotcom/liferay-portal/pull/35957
